onedal/basic_statistics/tests/test_basic_statistics.py

```diff
@@ -98,3 +107,65 @@ def test_option_weighted(queue, option, dtype):
 
         tol = fp32tol if res.dtype == np.float32 else fp64tol
         assert_allclose(gtr, res, rtol=tol)
+
+    @pytest.mark.skipif(not is_random_array_supported(), reason="requires scipy>=1.12.0")
+    @pytest.mark.parametrize("queue", get_queues())
+    @pytest.mark.parametrize("dtype", [np.float32, np.float64])
+    def test_basic_csr(queue, dtype):
+        seed = 42
+        s_count, f_count = 5000, 3008
+
+        gen = np.random.default_rng(seed)
+
+        from scipy.sparse import random_array
+
+        data = random_array(
+            shape=(s_count, f_count),
+            density=0.01,
+            format="csr",
+            dtype=dtype,
+            random_state=gen,
+        )
+
+        alg = BasicStatistics(result_options="mean")
+        res = alg.compute(data, queue=queue)
+
+        res_mean = res["mean"]
+        gtr_mean = data.mean(axis=0)
+        tol = 2e-5 if res_mean.dtype == np.float32 else 1e-7
```

Comment:
Is the tol values to match the other tests for BasicStatistics? Just curious to see if the tolerance between sparse and dense are equivalent or not for fp32 (since I've always believed 2e-5 is a bit high to be honest), and given the options_and_tests_csr implementation.